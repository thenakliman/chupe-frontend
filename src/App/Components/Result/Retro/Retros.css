.retro-container {
    margin: 1rem 18rem;
}

.create-retro-container {
    position: fixed;
    border: 1px solid #ccc;
    margin: 3% 9%;
}

.retro-button-container {
  display: flex;
  flex-direction: column;
  background: #fff;
  padding: 2rem;
}

.retro-row {
  cursor: pointer;
}

.retro-table-body tr:hover {
  background-color: #eff5ff;
  transition: background-color 0.5s ease;
}

.retro-creation-input {
  height: 2.8rem;
  width: 100%;
  font-size: 1.5rem;
  margin: 3rem 0;
}

.retro-button {
    padding: 1rem 2rem;
    margin: 2rem 0;
    float: right;
    color: white;

    cursor: pointer;
}

.create-retro-button {
    float: right;
    color: white;

    cursor: pointer;
    padding: 0.8rem 0.8rem;
    margin: 3rem 2rem;
    width: 9rem;
}

.create-retro-buttons-container {
    margin: 1rem 11rem;
}

.maximum-vote-count {
  height: 2.8rem;
  margin: 3rem 19rem;
  width: 5rem;
}