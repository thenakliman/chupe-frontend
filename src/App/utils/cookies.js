import Cookies from 'universal-cookie';
import {verify, decode} from 'jsonwebtoken';

const TOKEN_COOKIE_KEY = 'token';
const TOKEN_EXPIRY_COOKIE_KEY = 'expiryTime';
const TOKEN_USERNAME_KEY = 'username';
const audience = 'chupe-frontend';
const issuer = 'chupe';
export let cookies = new Cookies();
const Config = require('Config');

const setCookie = (key, value) => {
  cookies.set(key, value);
};

export const setCookies = (token) => {
  verify(token,
         Config.signingKey,
         {
            algorithms: ['HS512'],
            audience: audience,
            issuer: issuer,
            ignoreExpiration: false,
         },
         (e) => console.log(e)
  );

  const decodedToken = decode(token);
  setToken(token);
  setTokenExpiryTime(decodedToken.exp);
  setUsername(decodedToken.username);
};

const setToken = (token) => {
  setCookie(TOKEN_COOKIE_KEY, token);
};

const setTokenExpiryTime = (expiryTime) => {
  setCookie(TOKEN_EXPIRY_COOKIE_KEY, expiryTime);
};

const setUsername = (userName) => {
  setCookie(TOKEN_USERNAME_KEY, userName);
};

export const getUsername = () => {
  return cookies.get(TOKEN_USERNAME_KEY);
};

export const getToken = () => {
  if (!cookies.get(TOKEN_COOKIE_KEY) ||
      !cookies.get(TOKEN_EXPIRY_COOKIE_KEY) ||
      cookies.get(TOKEN_EXPIRY_COOKIE_KEY) < Date.now()) {
    return null;
  }

  return cookies.get(TOKEN_COOKIE_KEY);
};
