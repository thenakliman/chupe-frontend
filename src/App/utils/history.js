import {createHashHistory} from 'history';

export let history = createHashHistory();
