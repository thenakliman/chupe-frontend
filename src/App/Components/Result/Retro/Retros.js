import React from 'react';
import propTypes from 'prop-types';
/* eslint-disable */
import {CreateRetroPopUp} from './CreateRetroPopUp';
/* eslint-enable */
import {history} from '../../../utils/history';

require('./Retros.css');

export class Retros extends React.Component {
    constructor(props) {
      super(props);
      this.state = {creatingRetro: false};
      this.toggleCreatingRetro = this.toggleCreatingRetro.bind(this);
      this.createRetro = this.createRetro.bind(this);
    }

    componentDidMount() {
      this.props.getAllRetros();
    }

    toggleCreatingRetro() {
      this.setState({creatingRetro: !this.state.creatingRetro});
    }

    viewRetro(id) {
      history.push('/retro/' + id);
    }

    createRetro(retro) {
      this.props.createRetro(retro);
      this.toggleCreatingRetro();
    }

    render() {
      return (
      <div id='retro-container-id' className='retro-container'>
          <button id='retro-button-id'
            className='retro-button'
            type='button'
            onClick={this.toggleCreatingRetro}
          >
            Create Retro
          </button>
          {this.state.creatingRetro &&
           <CreateRetroPopUp
              createRetro={(retro) => this.createRetro(retro)}
              closeCreatePopUp={this.toggleCreatingRetro}
           />
           }
          <div>
            <table id='all-retros-ordered-list'>
              <thead>
                <tr className='retro-table-header'>
                  <th> Id </th>
                  <th> Name </th>
                  <th> Created By </th>
                  <th> Maximum Vote </th>
                </tr>
              </thead>
              <tbody id='all-retro-table-body-id'>
              {
                this.props.retros.map((retro) =>(
                  <tr key={`${retro.id}`}>
                    <td> {retro.id} </td>
                    <td className='retro-row'
                        id={`retro-row-id-${retro.id}`}
                        onClick={() => this.viewRetro(retro.id)}
                    >
                            {retro.name}
                    </td>
                    <td> {retro.createdBy}</td>
                    <td> {retro.maximumVote}</td>
                  </tr>
                ))
              }
              </tbody>
            </table>
          </div>
      </div>
      );
    }
}

Retros.propTypes = {
  retros: propTypes.array,
  getAllRetros: propTypes.func.isRequired,
  createRetro: propTypes.func.isRequired,
};
