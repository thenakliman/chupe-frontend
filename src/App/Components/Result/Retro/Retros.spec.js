/* eslint-disable */
import React from 'react';
import {Retros} from './Retros';
/* eslint-enable */
import {mount, shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import * as history from '../../../utils/history';

const retros = [{
  name: 'retro - 1',
  id: 10,
  createdBy: 'fake user - 1',
}, {
  name: 'retro - 2',
  id: 11,
  createdBy: 'fake user - 2',
}];

describe('Show retro component', () => {
  it('should call getAllRetros on component did mount ', () => {
    const getAllRetros = jest.fn();
    shallow(
        <Retros
            retros={retros}
            getAllRetros={getAllRetros}
        />
    );

    expect(getAllRetros).toHaveBeenCalledWith();
  });

  it('should have initial state for creatingRetro to false', () => {
    const getAllRetros = jest.fn();
    const wrapper = shallow(
        <Retros
            retros={retros}
            getAllRetros={getAllRetros}
        />
    );

    expect(wrapper.state().creatingRetro).toBe(false);
  });

  it('should set creatingRetro to true on click of create retro button', () => {
    const getAllRetros = jest.fn();
    const wrapper = shallow(
        <Retros
            retros={retros}
            getAllRetros={getAllRetros}
        />
    );

    wrapper.find('#retro-button-id').simulate('click');
    expect(wrapper.state().creatingRetro).toBe(true);
  });

  it('should call create retro', () => {
    const getAllRetros = jest.fn();
    const createRetro = jest.fn();
    const wrapper = shallow(
        <Retros
            retros={retros}
            createRetro={createRetro}
            getAllRetros={getAllRetros}
        />
    );

    const retro = {name: 'retro'};
    wrapper.instance().createRetro(retro);
    expect(createRetro).toHaveBeenCalledWith(retro);
    expect(wrapper.state().creatingRetro).toBe(true);
  });

  it('should call create retro on click of button', () => {
    const getAllRetros = jest.fn();
    const createRetro = jest.fn();
    const wrapper = mount(
        <Retros
            retros={retros}
            createRetro={createRetro}
            getAllRetros={getAllRetros}
        />
    );

    wrapper.find('#retro-button-id').simulate('click');
    expect(wrapper.find('CreateRetroPopUp').length).toBe(1);
    const testName = 'testName';
    wrapper.find('#create-retro-name').simulate(
        'change', {target: {value: testName}});

    const maximumVote = 29;
    wrapper.find('#create-retro-maximum-vote-count').simulate(
        'change', {target: {value: maximumVote}});

    wrapper.find('#create-retro-button-id').simulate('click');
    expect(createRetro).toHaveBeenCalledWith({
      name: testName,
      maximumVote: maximumVote,
    });
  });

  it('should render CreateRetroPopUp when creatingRetro is true', () => {
    const getAllRetros = jest.fn();
    const createRetro = jest.fn();
    const wrapper = mount(
        <Retros
            retros={retros}
            createRetro={createRetro}
            getAllRetros={getAllRetros}
        />
    );

    wrapper.setState({creatingRetro: true});
    expect(wrapper.find('CreateRetroPopUp').length).toBe(1);
  });

  it('should toggle state of creatingRetro on after creation of retro', () => {
    const getAllRetros = jest.fn();
    const createRetro = jest.fn();
    const wrapper = shallow(
        <Retros
            retros={retros}
            createRetro={createRetro}
            getAllRetros={getAllRetros}
        />
    );

    const retro = {name: 'retro'};
    wrapper.instance().createRetro(retro);
    expect(wrapper.state().creatingRetro).toBe(true);
  });

  it('should show row for each retro', () => {
    const getAllRetros = jest.fn();
    const wrapper = shallow(
        <Retros
            retros={retros}
            getAllRetros={getAllRetros}
        />
    );

    expect(wrapper.find('#all-retro-table-body-id').children().length).toBe(2);
  });


  it('should show retro on click of retro point', () => {
    const getAllRetros = jest.fn();
    history.history = {push: jest.fn()};
    const getPracticesAssessment = jest.fn(() => Promise.resolve());
    const wrapper = shallow(
        <Retros
            match={{params: {id: 11}}}
            retros={retros}
            getAllRetros={getAllRetros}
            getPracticesAssessment={getPracticesAssessment}
        />
    );

    wrapper.find('#retro-row-id-11').simulate('click');
    expect(history.history.push).toHaveBeenCalledWith('/retro/11');
  });

  describe('should match snapshot', () => {
    const wrapper = shallow(<Retros
        retros={retros}
        getAllRetros={jest.fn()}
    />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
