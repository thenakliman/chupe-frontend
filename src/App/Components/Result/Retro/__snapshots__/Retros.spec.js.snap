// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="retro-container"
  id="retro-container-id"
>
  <button
    className="retro-button"
    id="retro-button-id"
    onClick={[Function]}
    type="button"
  >
    Create Retro
  </button>
  <div>
    <table
      id="all-retros-ordered-list"
    >
      <thead>
        <tr
          className="retro-table-header"
        >
          <th>
             Id 
          </th>
          <th>
             Name 
          </th>
          <th>
             Created By 
          </th>
          <th>
             Maximum Vote 
          </th>
        </tr>
      </thead>
      <tbody
        className="retro-table-body"
        id="all-retro-table-body-id"
      >
        <tr
          className="selectable-row"
          id="retro-row-id-10"
          key="10"
          onClick={[Function]}
        >
          <td>
             
            10
             
          </td>
          <td
            className="retro-row"
          >
            retro - 1
          </td>
          <td>
             
            fake user - 1
          </td>
          <td>
             
          </td>
        </tr>
        <tr
          className="selectable-row"
          id="retro-row-id-11"
          key="11"
          onClick={[Function]}
        >
          <td>
             
            11
             
          </td>
          <td
            className="retro-row"
          >
            retro - 2
          </td>
          <td>
             
            fake user - 2
          </td>
          <td>
             
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
